app/src/main/kotlin/co/netguru/baby/monitor/client/feature/settings/ClientSettingsFragment.kt

```diff
@@ -93,13 +118,45 @@ class ClientSettingsFragment : BaseFragment() {
                     R.color.alpha_accent, R.drawable.ic_select_photo_camera
                 )
             }
+            checkVoiceAnalysisOption(resolveOption(child.voiceAnalysisOption))
         }
         configurationViewModel.resetState.observe(viewLifecycleOwner, Observer { resetState ->
             when (resetState) {
-                is ResetState.InProgress -> setupResetButton(true)
-                is ResetState.Failed -> setupResetButton(false)
+                is ChangeState.InProgress -> setupResetButton(true)
+                is ChangeState.Failed -> setupResetButton(false)
             }
         })
+
+        configurationViewModel.voiceAnalysisOptionState.observe(
+            viewLifecycleOwner,
+            Observer { voiceAnalysisChangeState ->
+                setupVoiceAnalysisRadioButtons(voiceAnalysisChangeState)
+            })
+    }
+
+    private fun setupVoiceAnalysisRadioButtons(voiceAnalysisChangeState: Pair<ChangeState, VoiceAnalysisOption?>) {
+        machineLearningOption.isEnabled = voiceAnalysisChangeState.first != ChangeState.InProgress
+        noiseDetectionOption.isEnabled = voiceAnalysisChangeState.first != ChangeState.InProgress
+        voiceAnalysisChangeState.second?.run {
+            checkVoiceAnalysisOption(resolveOption(this))
+        }
+    }
+
+    private fun resolveOption(
+        voiceAnalysisOption: VoiceAnalysisOption
+    ): Int {
+        return when (voiceAnalysisOption) {
+            VoiceAnalysisOption.MachineLearning -> R.id.machineLearningOption
+            VoiceAnalysisOption.NoiseDetection -> R.id.noiseDetectionOption
+        }
+    }
+
+    private fun checkVoiceAnalysisOption(optionToSet: Int) {
+        voiceAnalysisRadioButtons.apply {
+            setOnCheckedChangeListener(null)
```

Comment:
Is this one needed? This is setter, so `setOnCheckedChangeListener(voiceAnalysisCheckChangedListener())` should be enough.

Reply:
i need to nullify it in order to programmatically check the buttons without triggering the listener